common/azure-common/src/main/java/com/azure/common/exception/TooManyRedirectsException.java

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.common.exception;
+
+import com.azure.common.http.HttpResponse;
+
+/**
+ * This exception is thrown when an HTTP request has reached the maximum number of redirect attempts
+ * with HTTP status code of 5XX.
```

Comment:
The javadoc for this is incorrect, redirect attempts are 3xx status codes and not 5xx.

I don't think having the HttpResponse is useful in this case. HttpRequest may be what is interesting because you may want to understand. ex. "what caused us to be redirected so many times?" or "how many redirects are allowed?"

Reply:
Fixed the common.

For HttpRequest and limits, we think instead of take in httpRequest or limits as parameter, we also can get the url from response().request().url(). Let sdk client to handle the info based on client needs.

We agreed not putting additional para into exception for now.